servant-server/src/Servant/Server/Internal/RoutingApplication.hs

```diff
@@ -103,6 +105,10 @@ data Delayed env c where
              , authD     :: DelayedIO auth
              , bodyD     :: DelayedIO body
              , serverD   :: captures -> auth -> body -> Request -> RouteResult c
+             , cleanupD  :: body -> IO ()
```

Comment:
1/ Why is it more intuitive? Clean up will rarely (if ever) need to look at the raw request.

2/ In my case, I get filepaths of temp files and what not by looking at `body`. We can indeed add other arguments to cleanupD, but not make it a simple action, because I otherwise don't have any mean for my cleanup thing to get its hands on the temp files it is supposed to delete. More generally, this feature is useful anytime we want to clean up after body checks and other stuffs (anything, really). Except maybe for somewhat "global" stuffs, we need a way to for the clean up action to get its hands on whatever it is that needs cleaning up, so I use an argument for that.

I would not mind adding more arguments (results of the other checks). But making it a `DelayedIO` so that I have to parse everything from the request again? No.